server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -254,6 +254,35 @@ public Pool find(Serializable id) {
         return listByCriteria(crit, pageRequest, postFilter);
     }
 
+    /**
+     * Determine if owner has at least one active pool
+     *
+     * @param o Owner whose subscriptions should be inspected.
+     * @param date The date to test the active state.
+     *        Set to null for current.
+     * @return boolean is active on test date.
+     */
+    @SuppressWarnings("unchecked")
+    @Transactional
+    public boolean hasActiveEntitlementPools(Owner o, Date date) {
+
+        if (o == null) {
+            return false;
+        }
+        if (date == null) {
+            date = new Date();
+        }
+        Criteria crit = createSecureCriteria();
+        crit.add(Restrictions.eq("activeSubscription", Boolean.TRUE));
+        crit.add(Restrictions.eq("owner", o));
+        crit.add(Restrictions.le("startDate", date));
+        crit.add(Restrictions.ge("endDate", date));
+        crit.setProjection(Projections.rowCount());
+
+        long count = (Long) crit.list().get(0);
```

Comment:
I've always done this as follows, however, I'm not sure if there is a difference in using uniqueResult vs list().get(0). Might be something to investigate -- I'm not 100% sure which is better TBH.

```java
long count = (Long) crit.uniqueResult();
```